### PR TITLE
Replace hackney with httpc

### PR DESCRIPTION
**Pull request by whatyouhide**
This is the start of the work proposed in https://github.com/parroty/excoveralls/pull/231.

The main missing thing is Security™, that is, I need to add all the right SSL options to make the `:httpc` request secure. I'll work on that this afternoon if there is interest in this, @parroty?

Btw @parroty, also feel free to send me an email (email is on my GitHub profile) or write me a DM [on Twitter](https://twitter.com/whatyouhide) to chat about helping out with maintaining this library 🙃

### CODE REVIEW

**Comment by whatyouhide** [comment]
@parroty ping 🙃 Would love to get an email and chat! 😉

**Comment by whatyouhide** [comment]
@parroty ping 🏓 🙃

**Comment by parroty** [comment]
Sorry being slow to respond. I am seeing actions errors, but haven't been able look into it. Do you have any idea?

**Comment by whatyouhide** [comment]
@parroty let's start out with https://github.com/parroty/excoveralls/pull/316 😄

**Comment by whatyouhide** [comment]
@parroty finally fixed tests in CI on the supported versions, so this is **ready to merge**!!! Exciting times ahead. 💟

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -25,8 +25,8 @@ jobs:
       - uses: actions/cache@v3
         with:
          path: deps
-         key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
+         key: ${{ runner.os }}-${{ matrix.elixir }}-otp${{ matrix.otp }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
```
Comment (whatyouhide) [inline .github/workflows/tests.yml:28]: @parroty having a single cache for all OTP versions causes issues when testing across different versions. Most projects I've used actually go with a per-version cache, so this is what this change is.

**Comment by whatyouhide** [comment]
Tested this out in [Xandra](https://github.com/lexhide/xandra), works good 🙃

**Comment by whatyouhide** [comment]
Woohooo! Thanks @parroty. Looking forward to a release 😉

File: lib/excoveralls/poster.ex [hunk 2]
```diff
@@ -22,44 +17,129 @@ defmodule ExCoveralls.Poster do
     end
   end
 
-  defp send_file(file_name, options) do
-    Application.ensure_all_started(:hackney)
+  defp upload_zipped_json(content, options) do
+    Application.ensure_all_started(:ssl)
+    Application.ensure_all_started(:httpc)
```
Comment (gmile) [inline lib/excoveralls/poster.ex:22]: Out of curiosity, why does `:httpc` needs to be started explicitly? Starting `:inets.start()` and `:ssl.start()` typically was sufficient for me in the past 🤔

Comment (whatyouhide) [reply lib/excoveralls/poster.ex:22]: It probably is sufficient, I just overlooked this. PRs welcome 😉

Comment (parroty) [reply lib/excoveralls/poster.ex:22]: Thanks for the comment. Do you see specific negative implication for this extra start logic? (Just a small overhead).

I'll be publishing the release later today (after going back home), but wondering whether to update this before the release.

Comment (whatyouhide) [reply lib/excoveralls/poster.ex:22]: @parroty `httpc` is not an app. Just merge https://github.com/parroty/excoveralls/pull/318 before releasing 😉

**Pull request merged**